cli/tools/upgrade.rs

```diff
@@ -378,8 +378,17 @@ pub async fn upgrade(
   let factory = CliFactory::from_flags(flags).await?;
   let client = factory.http_client();
   let current_exe_path = std::env::current_exe()?;
-  let output_exe_path =
-    upgrade_flags.output.as_ref().unwrap_or(&current_exe_path);
+  let full_path_output_flag = if let Some(output) = upgrade_flags.output {
+    let output = PathBuf::from(output);
+    if output.is_relative() {
+      Some(factory.cli_options().initial_cwd().join(output))
+    } else {
+      Some(output)
+    }
```

Comment:
I think we can jsut do this:

```suggestion
    factory.cli_options().initial_cwd().join(output)
```